### PR TITLE
Improve logging around incoming messages.

### PR DESCRIPTION
**Pull request by charlesmchen**
PTAL @michaelkirk

### CODE REVIEW

**Review by michaelkirk — COMMENTED** [review]
LGTM

File: src/Messages/TSMessagesManager.m [hunk 1]
```diff
@@ -914,12 +916,12 @@ - (TSIncomingMessage *_Nullable)handleReceivedEnvelope:(OWSSignalServiceProtosEn
                                                              messageBody:body
                                                            attachmentIds:attachmentIds
                                                         expiresInSeconds:dataMessage.expireTimer];
+          DDLogDebug(@"%@ incoming 1:1 text message: %@", self.tag, incomingMessage.debugDescription);
+          [incomingMessage saveWithTransaction:transaction];
```
Comment (michaelkirk) [inline src/Messages/TSMessagesManager.m:920]: Why was this moved? Was there something wrong with the previous logic?

File: src/Messages/TSMessagesManager.m [hunk 2]
```diff
@@ -914,12 +916,12 @@ - (TSIncomingMessage *_Nullable)handleReceivedEnvelope:(OWSSignalServiceProtosEn
...
           thread = cThread;
       }
 
       if (thread && incomingMessage) {
-          [incomingMessage saveWithTransaction:transaction];
```
Comment (charlesmchen) [inline src/Messages/TSMessagesManager.m:921]: See here.

File: src/Messages/TSMessagesManager.m [hunk 3]
```diff
@@ -892,7 +894,7 @@ - (TSIncomingMessage *_Nullable)handleReceivedEnvelope:(OWSSignalServiceProtosEn
                                                                      messageBody:body
                                                                    attachmentIds:attachmentIds
                                                                 expiresInSeconds:dataMessage.expireTimer];
-
+                  DDLogDebug(@"%@ incoming group text message: %@", self.tag, incomingMessage.debugDescription);
                   [incomingMessage saveWithTransaction:transaction];
```
Comment (charlesmchen) [inline src/Messages/TSMessagesManager.m:898]: See here.

File: src/Messages/TSMessagesManager.m (hunk 1 above)
Comment (charlesmchen) [reply src/Messages/TSMessagesManager.m:920]: We were saving messages twice in some cases.  See comments above and below.  We should either save once when the message is created (as I've done here) or once later.  I figured the latter was a little easier to ensure to be correct.

Comment (michaelkirk) [reply src/Messages/TSMessagesManager.m:920]: Got it. Thanks
